.github/workflows/test.yml

```
@@ -8,7 +8,7 @@ on:
 
 jobs:
   test:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-24.04
```

Python 3.7 isn't provided in Ubuntu 24.04 GitHub runner.

```suggestion
    runs-on: ubuntu-22.04
```

Consider dropping Py37 support later.